exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/Foreman.java

```diff
@@ -296,7 +295,7 @@ public void run() {
        * would wait on the cancelling thread to signal a resume and the cancelling thread would wait on the Foreman
        * to accept events.
        */
-      acceptExternalEvents.countDown();
+      stateSwitch.start();
```

Comment:
that is true indeed, I didn't notice the Foreman was also using the event queue before it counted down the latch. I will update the fix to respect this behavior